operators/src/mock/mock_feature_collection_source.rs

```diff
@@ -0,0 +1,267 @@
+
+    ($geometry:ty, $output:ident, $newtype:ident) => {
+        type $newtype = MockFeatureCollectionSource<$geometry>;
+
+        #[typetag::serde]
+        impl VectorOperator for $newtype {
+            fn initialize(
+                self: Box<Self>,
+                context: ExecutionContext,
+            ) -> Result<Box<InitializedVectorOperator>> {
+                InitializedOperatorImpl::create(
+                    self.params,
+                    context,
+                    |_, _, _, _| Ok(()),
+                    |_, _, _, _, _| {
+                        Ok(VectorResultDescriptor {
+                            data_type: <$geometry>::DATA_TYPE,
+                            projection: Projection::wgs84().into(), // TODO: get from `FeatureCollection`
+                        })
+                    },
+                    vec![],
+                    vec![],
+                )
+                .map(InitializedOperatorImpl::boxed)
+            }
+        }
+
+        impl InitializedOperator<VectorResultDescriptor, TypedVectorQueryProcessor>
+            for InitializedOperatorImpl<
+                MockFeatureCollectionSourceParams<$geometry>,
+                VectorResultDescriptor,
+                (),
+            >
+        {
+            fn query_processor(&self) -> Result<TypedVectorQueryProcessor> {
+                Ok(TypedVectorQueryProcessor::$output(
+                    MockFeatureCollectionSourceProcessor {
+                        collection: self.params.collection.clone(),
+                    }
+                    .boxed(),
+                ))
+            }
+        }
+    };
+}
+
+impl_mock_feature_collection_source!(NoGeometry, Data);
+impl_mock_feature_collection_source!(MultiPoint, MultiPoint);
+impl_mock_feature_collection_source!(MultiLineString, MultiLineString);
+impl_mock_feature_collection_source!(MultiPolygon, MultiPolygon);
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use futures::executor::block_on_stream;
+    use geoengine_datatypes::collections::MultiPointCollection;
+    use geoengine_datatypes::primitives::{BoundingBox2D, Coordinate2D, FeatureData, TimeInterval};
+
+    #[test]
+    fn serde() {
+        let collection = MultiPointCollection::from_data(
+            MultiPoint::many(vec![(0.0, 0.1), (1.0, 1.1), (2.0, 3.1)]).unwrap(),
+            vec![TimeInterval::new_unchecked(0, 1); 3],
+            [(
+                "foobar".to_string(),
+                FeatureData::NullableDecimal(vec![Some(0), None, Some(2)]),
+            )]
+            .iter()
+            .cloned()
+            .collect(),
+        )
+        .unwrap();
+
+        let source = MockFeatureCollectionSource {
+            params: MockFeatureCollectionSourceParams { collection },
+        }
+        .boxed();
+
+        let serialized = serde_json::to_string(&source).unwrap();
+        let collection_bytes = [
+            65, 82, 82, 79, 87, 49, 0, 0, 144, 1, 0, 0, 16, 0, 0, 0, 0, 0, 10, 0, 14, 0, 12, 0, 11,
```

Comment:
why are there so many 0s?

Reply:
Empty Buffers?… I don't know and what shall I address here?

Reply:
check the validity of the serialization result? Why does it take 1.4kB to save 3 features?

Reply:
Well, it is a workaround that uses the IPC format (https://arrow.apache.org/docs/format/Columnar.html#serialization-and-interprocess-communication-ipc) to have some kind of serialization.
I guess it is obvious that it is far from ideal, but at least it works for now.